jekyll/_cci2/use-the-cli-for-config-policy-management.adoc

```diff
@@ -0,0 +1,226 @@
+
+. Create a directory for your `rego` policy files, for example:
++
+[source,shell]
+----
+mkdir ./config-policies
+----
+
+- Set up a `.circleci/config.yml` file for your policies repository. This example will push policies to CircleCI on commits to `main` and show a diff of the policy bundle on commits to all other branches:
++
+[source,yaml]
+----
+version: 2.1
+
+orbs:
+  circleci-cli: circleci/circleci-cli@0.1.9
+
+workflows:
+  main-workflow:
+    jobs:
+      - diff-policy-bundle:
+          context: [ security-operations ]
+          filters:
+            branches:
+              ignore: main
+      - push-policy-bundle:
+          context: [ security-operations ]
+          filters:
+            branches:
+              only: main
+
+jobs:
+  diff-policy-bundle:
+    executor: circleci-cli/default
+    resource_class: small
+    steps:
+      - checkout
+      - run:
+          name: Diff policy bundle
+          command: circleci policy diff ./config --owner-id $OWNER_ID
+
+  push-policy-bundle:
+    executor: circleci-cli/default
+    resource_class: small
+    steps:
+      - checkout
+      - run:
+          name: Push policy bundle
+          command: circleci policy push ./config --no-prompt --owner-id $OWNER_ID
+----
+
+Let us break down the previous config:
+
+The following orb makes the `circleci-cli/default` executor available to our jobs
+
+```yaml
+orbs:
+  circleci-cli: circleci/circleci-cli@0.1.9
+```
+
+We then declare two jobs: `diff-policy-bundle` and `push-policy-bundle` to run the policy diff and push commands respectively.
+
+Note that `$OWNER_ID` is an environment variable setup in project settings that is simply your organization id.
+Your organization id is a uuid value that can be found on the organization settings page. 
+
+```yaml
+jobs:
+  diff-policy-bundle:
+    executor: circleci-cli/default
+    resource_class: small
+    steps:
+      - checkout
+      - run:
+          name: Diff policy bundle
+          command: circleci policy diff ./config --owner-id $OWNER_ID
+
+  push-policy-bundle:
+    executor: circleci-cli/default
+    resource_class: small
+    steps:
+      - checkout
+      - run:
+          name: Push policy bundle
+          command: circleci policy push ./config --no-prompt --owner-id $OWNER_ID
+```
+
+We declare a workflow to run the diff job when not on branch `main` and the push job only on branch `main`
+
+```yaml
+workflows:
+  main-workflow:
+    jobs:
+      - diff-policy-bundle:
+          context: [ security-operations ]
+          filters:
+            branches:
+              ignore: main
+      - push-policy-bundle:
+          context: [ security-operations ]
+          filters:
+            branches:
+              only: main
+```
+
+Note the context for each job: `security-operations`. This context name is arbitrary, however a context is needed to authenticate the CLI. The context must declare an environment variable `CIRCLECI_CLI_TOKEN` that will be used by the CLI.
```

Comment:
How come a Context is required? could it not just be a project env var added via the app?

Reply:
@davidmparrott

Reply:
I'm not sure I understand the question. Here we are trying to say that you have to have this env var present, and it has to be named this way. As I understand it, contexts are how we inject environment variables into a build environment.

Reply:
I guess all we need is CIRCLECI_CLI_TOKEN environment variable defined, with proper value.
Other ways of defining the variable would also work, eg project level variable, or defining withing the config(extremely unsafe).
We found defining the variable in a 'restricted' context (restricted access to some users / group of users) the safest of all, and that's why the recommendation.

Reply:
Thank you!